mathesar/imports/excel.py

```diff
@@ -28,10 +28,28 @@ def insert_records_from_dataframe(name, schema, column_names, engine, comment, d
     return table
 
 
+def remove_empty_rows_and_columns_from_dataframe(df):
+    if df.iloc[0].isna().any():
+
+        # drop rows with all NaN values
+        df.dropna(how='all', inplace=True)
+
+        # drop columns with all NaN values
+        df.dropna(axis=1, how='all', inplace=True)
+
+    if all(df.columns.str.startswith('Unnamed')):
+        df.columns = df.iloc[0]
+        df = df[1:]
```

Comment:
I think there should be a conderation here for whether or not the first row of the data is a header in #3030.

Reply:
Yeah, that is planned for the coming PRs.